### DISCUSSION TITLE
Can pandoc preserve whitespace when converting RTF to MD?

### DISCUSSION THREAD

**Discussion by GeoffreyPlitt** [jgm/pandoc#10335] [General]
I'm using Pandoc to convert an RTF doc to an MD for a musical chord chart. The whitespace is important, because the chords need to align with the lyrics. I've attached the example RTF (input) and MD (output) that I got, using this command:

`cat chart.rtf | pandoc --from=rtf --to=markdown > out.md`

The whitespace is not preserved correctly. Adding "--wrap=preserve" does not make a difference. Is there a way to preserve the whitespace?
[chart.rtf.zip](https://github.com/user-attachments/files/17511996/chart.rtf.zip)
[out.md](https://github.com/user-attachments/files/17511997/out.md)

**Comment by badumont** [comment]
Maybe you could replace all whitespaces with non-breakable spaces? However, another problem with this kind of formatting is that the exact placement of the letters will depend on the font used…

**Reply by GeoffreyPlitt** [reply]
Font-wise, any monospaced font will work. The alignment was done in RTF with a monospaced font. So font is not the issue.
Non-breaking spaces sounds interesting... except this isn't a line-wrap issue. This is an issue of whitespace being deleted.

**Comment by jgm** [comment]
The RTF reader attempts to figure out what is a "code block" and will translate that into a code block (with spacing preserved).  Everything else will be treated as reflowable paragraph content.

The way it detects code is by looking at the font.

**Comment by badumont** [comment]
> Non-breaking spaces sounds interesting... except this isn't a line-wrap issue.

I don't understand what “except” implies here. You can try it yourself:

pandoc -f rtf -t markdown <<< 'normal    spaces'

normal spaces

pandoc -f rtf -t markdown <<< 'non-breaking      spaces'

non-breaking      spaces

**Reply by GeoffreyPlitt** [reply]
But this doesn't preserve the exact number of spaces from the source document.

**Comment by bpj** [comment]
If editing the RTF in a "word processor" is an option might replacing it with a table be an option?

**Reply by GeoffreyPlitt** [reply]
A bit confused, how are tables related to my question?

**Reply by bpj** [reply]
You use a two-row table (without visible borders) with the chords in the upper row and the words in the lower row with one column for each pair of chords and words which should line up with each other using as many columns as necessary to make them line up properly. That way you don't need to use a fixed width font. I use that trick for linguistic interlinear glosses which have a similar line-up requirement.
